restapi.go

```diff
@@ -1311,7 +1311,52 @@ func (s *Session) ChannelMessageSendComplex(channelID string, data *MessageSend)
 		data.Embed.Type = "rich"
 	}
 
-	response, err := s.RequestWithBucketID("POST", EndpointChannelMessages(channelID), data, EndpointChannelMessages(channelID))
+	var response []byte
+	if data.File != nil {
+		body := &bytes.Buffer{}
+		bodywriter := multipart.NewWriter(body)
+
+		// What's a better way of doing this? Reflect? Generator? I'm open to suggestions
+
+		if err = bodywriter.WriteField("content", data.Content); err != nil {
+			return
+		}
+
+		var embed []byte
+		embed, err = json.Marshal(data.Embed)
+		if err != nil {
+			return
+		}
+		err = bodywriter.WriteField("embed", string(embed))
+		if err != nil {
+			return
+		}
+
+		if err = bodywriter.WriteField("tts", strconv.FormatBool(data.Tts)); err != nil {
+			return
+		}
+
+		var writer io.Writer
+		writer, err = bodywriter.CreateFormFile("file", data.File.Name)
+		if err != nil {
+			return
+		}
+
+		_, err = io.Copy(writer, data.File.Reader)
+		if err != nil {
+			return
+		}
+
+		err = bodywriter.Close()
+		if err != nil {
+			return
+		}
+
+		endpoint := EndpointChannelMessages(channelID)
```

Comment:
Sorry I'm being really pedantic here, could you move this up outside the if, and use it in the else below as well, I'm being all weird and finicky :)